### PR TITLE
feat: Add build instructions for 4.0.2

### PR DESCRIPTION
**Pull request by rhx**
Adding the swiftenv-build/4.0.2 file in this pull request adds support for building swift-4.0.2 from source.  This was tested on Ubuntu 16.04 (64 bit).

### CODE REVIEW

**Comment by SDGGiesbrecht** [comment]
Thanks @rhx. I was considering submitting these myself, but you beat me to it.

@kylef, what do you think?

**Comment by rhx** [comment]
It looks like the AppVeyor fail seems to be a setup problem that has nothing to do with the pull request.

File: share/swiftenv-install/4.0.3 [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  'ubuntu16.04' )
+    URL="https://swift.org/builds/swift-4.0.3-release/ubuntu1604/swift-4.0.3-RELEASE/swift-4.0.3-RELEASE-ubuntu16.04.tar.gz"
+    ;;
+
+  'ubuntu16.10' )
+    URL="https://swift.org/builds/swift-4.0.3-release/ubuntu1610/swift-4.0.3-RELEASE/swift-4.0.3-RELEASE-ubuntu16.10.tar.gz"
+    ;;
+
+  'ubuntu17.04' )
+    URL="https://swift.org/builds/swift-4.0.3-release/ubuntu1610/swift-4.0.3-RELEASE/swift-4.0.3-RELEASE-ubuntu16.10.tar.gz"
```
[the remark this answers is not in the document]
Comment (rhx) [reply share/swiftenv-install/4.0.3]: Yes, that's correct: while there are no pre-built binaries for 17.04 and 17.10, but the 16.10 ones are working fine under 17.*!

File: share/swiftenv-install/4.0.3 [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    ;;
+
+  'ubuntu17.10' )
+    URL="https://swift.org/builds/swift-4.0.3-release/ubuntu1610/swift-4.0.3-RELEASE/swift-4.0.3-RELEASE-ubuntu16.10.tar.gz"
```
[the remark this answers is not in the document]
Comment (rhx) [reply share/swiftenv-install/4.0.3]: Yes, that's correct: while there are no pre-built binaries for 17.04 and 17.10, but the 16.10 ones are working fine under 17.10!

**Pull request closed without merging**